cityofaustin/atd-moped
# Adds audit triggers to moped_proj_personnel and moped_proj_personnel_roles tables

## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/16149

## Testing
**URL to test:** 

local

**Steps to test:**
- run the sql command below and note the values for the audit fields (`created_by_user_id`, `created_at`, `updated_by_user_id`, and `updated_at`) for the project, personnel and personnel roles tables
- add a new team member to a project
- `updated_by_user_id` and `updated_at` should have updated for all records on the project table, and all audit fields should have updated for the new record you created on the personnel and personnel roles tables
- edit the new record's notes field and confirm that the `updated_at` and `updated_by_user_id` fields have been updated for the project and personnel tables
- go to the entry you created and updated its role by unchecking one and checking a different one
- `updated_by_user_id` and `updated_at` should have updated for all records on the project table, `updated_at` and `updated_by_user_id` should be updated for both new records on the personnel table, and all audit fields should be updated for the new record in the personnel roles table
- log out of moped and log back in using a different account
- go to the entry you created and updated its role by unchecking one and checking a different one
- `updated_by_user_id` and `updated_at` should have updated on the personnel roles records, along with the parent personnel records and the project table records

Query for moped_proj_personnel and moped_proj_personnel_roles
```sql
SELECT
	mp.project_id,
	mp.updated_by_user_id,
	mp.updated_at,
	mpp.project_personnel_id,
	mpp.created_at,
	mpp.updated_at,
	mpp.created_by_user_id,
	mpp.updated_by_user_id,
	mppr.project_role_id,
	mppr.created_at,
	mppr.updated_at,
	mppr.created_by_user_id,
	mppr.updated_by_user_id,
	mppr.is_deleted
FROM
	moped_project mp
	LEFT JOIN moped_proj_personnel mpp ON mp.project_id = mpp.project_id
	LEFT JOIN moped_proj_personnel_roles mppr ON mpp.project_personnel_id = mppr.project_personnel_id
WHERE
	mp.project_id = <your project id>;
```

---
#### Ship list
